JS/page-number-change.js

```diff
@@ -0,0 +1,22 @@
+function numberFieldChange() {
+    let val = document.getElementById("pageNumber").value;
+    pageCount = document.getElementById("pageNumber").value;
+    let signOfVal = Math.sign(val);
+    if (signOfVal == 1) {
+        if (val > results.items.length / 4) {
+            endIndex = results.items.length;
+            startIndex = endIndex - 4;
+            document.getElementById("pageNumber").value=results.items.length/4;
+            displayCards(results, startIndex, endIndex);
+        } else {
+            endIndex = 4 * val;
+            startIndex = endIndex - 4;
+            displayCards(results, startIndex, endIndex);
+        }
+
+    } else {
+        alert("Enter a valid page Number");
+        document.getElementById("pageNumber").value=1;
+    }
+
+}
```

Comment:
Extract 4 to constant and use. In that way, in the future, if you want to change the value you need to do lot of changes.